pywps/inout/inputs.py

```diff
@@ -99,7 +98,8 @@ def json(self):
             'data_format': self.data_format.json,
             'asreference': self.as_reference,
             'supported_formats': [frmt.json for frmt in self.supported_formats],
-            'file': self.file if self.prop == 'file' else None,
+            'file': self.file,
```

Comment:
The issue with this is that it's going to force a download of the URL locally. Look at the URLHandler `file` property getter in `inout/basic.py`. 
The url is already stored under the `href` key.

Reply:
would this work then?
` self.file if self.prop in ['file', 'url'] else None`

For the 'url' case (`UrlHandler`), the file is already downloaded at this point, and when the request is deserialized, no other download takes place... so we need the reference to the file property.

Reply:
The file property of `UrlHandler` starts like this:

```python

    @property
    def file(self):
        if self._file is not None:
            return self._file

        self._file = self._build_file_name(href=self.url)
        .....
```

Reply:
If the file is downloaded by URLHandler, then there's a bug.

In theory, the file is downloaded when the `file` property's getter is called. For the URLHandler, this should not happen unless explicitly requested by the Process handler. The point of the URLHandler is precisely to avoid downloading the file locally and let the process handler deal with data access. Our use case for this was netCDF files accessed through the opendap protocol. Users can send an opendap link to a process and have the process extract and process a few grid cells without having to download the entire file to the PyWPS server. 

Your fix would still force a download, which we want to avoid if `prop == 'url'`.